### ISSUE TITLE
I can't drag folders to the playlist

### ISSUE THREAD

**Issue by CPUhead** [OpenShot/openshot-qt#3492]
**Describe the bug:**
When I want to drag a folder with files I cannot and it gives me the symbol the mouse that if it is possible but I try and it does not leave me I have to select all the files and drag them it is very tedious to do it

**Steps to reproduce the behavior:**
1. Go to 'foldes'
2. drag on 'Playlist'
3. nothing happens does not respond

**Expected behavior:**

**System Details:**
- OpenShot Version [2.5.1-dev2
[libopenshot.log](https://github.com/OpenShot/openshot-qt/files/4638663/libopenshot.log)
[openshot-qt.log](https://github.com/OpenShot/openshot-qt/files/4638667/openshot-qt.log)
[openshot-qt.log](https://github.com/OpenShot/openshot-qt/files/4638668/openshot-qt.log)
[libopenshot.log](https://github.com/OpenShot/openshot-qt/files/4638673/libopenshot.log)

]: {{ log_context.openshot_qt_version }}
- Operating System / Distro: Linux Mint 19.3

**Log Files:**
If you are experiencing a crash, please collect and attach logs of the problem.
Log files can be found in your HOME/.openshot_qt/ folder. Please compress or zip
the log files (as they can become quite large).

**Exception / Stacktrace:**
If you are experiencing a crash or strange output on the command line, please
attach a small snippet of the exception here, but please keep it as small as
possible (for readability).

**Screenshots:** (Optional)
If applicable, add screenshots to help explain your problem. You can include screenshots by
copy/pasting them here, dragging-and-dropping into this form, or clicking below and loading
images saved to your computer. All images are public, so please don't post screenshots
containing personal information.

**Comment by ctsdownloads** [comment]
I don't believe this is a feature OpenShot currently offers.

**Comment by ferdnyc** [comment]
No, it isn't. Though, if/when the new files-model code merges, it probably wouldn't be very difficult to add. With the caveat that if the folder being dropped contains any _non_-media files, the user's going to have to dismiss a "Not a valid video, audio, or image file" popup for each one.

**Comment by ferdnyc** [comment]
This feature is now available in the latest Daily Builds, which are downloadable at https://openshot.org/download/. Testing and feedback would be greatly appreciated.

The caveats I listed with the PR are also very relevant, and should be kept in mind. Specifically:

> When processing drag-and-drop imports, OpenShot will recurse into any directories it finds, attempting to import all contained media. A few quirks of this mode:
> 
> 1. The import switches into a "silent running" mode, where the dialog box alerts about files that it can't import are turned off. **Any unrecognized media will just be silently ignored.**
> 1. **The import is totally flat ­— this doesn't add folders _to_ Project Files**, just lets you import all of the media _inside_ a folder (and possibly subfolders) into the list, which is still as flat as ever.
> 1. **I have no _idea_ what will happen if you drag a folder containing an `.osp` file in** with other media, but I imagine it won't be pretty. Just don't do that.
> 1. In general it's not even remotely bulletproof, nor is it intended to be. Kitchen-sink importing your entire media library on the off chance you might use something is still _not_ a supported way of working with OpenShot. Project Files is meant to hold only the media files that are _**actively in use**_ in the project (meaning, they have been placed on the Timeline). **PROJECT FILES IS A WORKSPACE, NOT A LIBRARY.** If you try to do dumb things with this, it will almost certainly break and it'll be your own fault.
> 

In addition, the new features brings with it yet _another_ large-scale reworking of the Image Sequence import handling, intended to address several bugs that were revealed while testing the new import code. Those changes apply to **all** imports, whether they're done via drag-and-drop, the File menu, or the Import Files toolbar button. The sorting change also affects _non_-image sequence imports.

1. **Image sequences are now limited to one per directory, per import** — whether you say Yes or No to the question about importing a sequence, no other sequences will be detected in that same directory for the remainder of the import "session".
1. The rewritten Image Sequence importing **should be more robust even in the typical non-folder import cases**. (That's what led to the new once-per-directory restriction, because it was doing all sorts of messed-up things when it tried to import a second one from the same path.)
1. **The list of files being imported is now sorted** before the import starts, because the results of Python's `os.walk` are all over the place and it was screwing up the image sequence imports to start from weird numbers.

Regarding the last item, I'm not entirely sure _how_ the list will be sorted when OpenShot is running localized to a language other than English. Because it's just using Python's `sorted()`, I suspect the sort order _may not_ correctly match the UI language... but I'm not really in a position to test that myself.

If the sorting is indeed wrong for non-English locales or non-Arabic writing systems, please report it by opening a new issue (be great if you could @-tag me as well); if necessary I can investigate switching over to sorting via Qt's QLocale.

But the sorting may not be a problem. The primary concern is making sure image sequence filenames are in _numerical_ order, and that should be the same regardless what language is used for the filenames and/or UI.

**Comment by ferdnyc** [comment]
Oh, and a caveat I edited out as it was mostly technical details, but still should've mentioned more explicitly:

### This only works for drag-and-drop imports.

The Open Files dialog (shown when you select **Files > Import Files...** or click the toolbar button) will not allow you to select folders for import, only files.

That appears to be a limitation of Qt's file-selection functionality, which strongly differentiates between a _file_ selection dialog and a _folder_ selection dialog in a way that seems to prevent mixed use. So far I haven't been able to overcome that limitation, so this doesn't work when using the file selection dialog. I would very much like to make it work, and I'll keep trying to find some way around that problem.

**Comment by USATechDude** [comment]
I've tested drag-and-dropping folders from my Debian Linux VM and my  MacBook Pro laptop using the [Media Test Files](https://s3.amazonaws.com/releases.openshot.org/testing/openshot-testing.zip).  I imported the entire contents of the zip file that I extracted to a folder.  On both systems, the import works correctly from the File Manager and the Finder.  I told instructed systems to import the animation images as a sequence which created snow%04d.png.  The README text file was ignore during the import.

**Comment by USATechDude** [comment]
@ferdnyc I just tested importing a folder with a subfolder to OpenShot on my Windows 10 system using the latest Daily Build (September 20, 2020).  Import was successful and OpenShot correctly identified the images in the animation folder as an image sequence.

![Screen Shot 2020-10-20 at 2 33 28 PM](https://user-images.githubusercontent.com/65924242/96635536-96178f00-12e1-11eb-96e4-e03a200bf942.png)

**Comment by USATechDude** [comment]
@ferdnyc I downloaded and installed [OpenShot-v2.5.1-dev2-1603218028-07a447c9-78db4b94-x86_64.dmg](https://github.com/OpenShot/openshot-qt/releases/download/daily/OpenShot-v2.5.1-dev2-1603218028-07a447c9-78db4b94-x86_64.dmg) (October 20, 2020) on my MacBook Pro laptop.  I tested the import folder function by dragging-and-dropping the Media Test Files folder to the Project List.  

Now, I can drag-and-drop single and multiple images, audio and video but not image sequences.  When I drag-and-drop an image sequence, I am prompted whether or not I want to import the file as a sequence, which I answer yes.  However, I cannot use the mouse to do so, I have to use the keyboard to tab to the Yes button then press Return.

When I use the File Import dialog the image sequence imports to the Project Files list and I can preview the Snow%04d.png file using the right-click menu.  When I add the image sequence to the timeline then press play under the Preview window, OpenShot either freezes at frame 13 or crashes.  If OpenShot freezes, I have to Force Quit the application.

Windows and Linux tested successfully without issues.
